localization/beluga_demo_bearing_localization/src/beluga_bmcl_demo.cpp

```diff
@@ -0,0 +1,491 @@
+// Copyright 2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <execution>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include <range/v3/view/concat.hpp>
+#include <range/v3/view/repeat.hpp>
+
+#include <sophus/se2.hpp>
+#include <sophus/se3.hpp>
+
+#include <beluga/beluga.hpp>
+#include <beluga/policies/every_n.hpp>
+#include <beluga/policies/on_motion.hpp>
+#include <beluga/policies/policy.hpp>
+#include <beluga/sensor/data/landmark_map.hpp>
+
+#include <beluga_ros/particle_cloud.hpp>
+#include <beluga_ros/tf2_sophus.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <message_filters/subscriber.h>
+
+#include <tf2/convert.h>
+#include <tf2/utils.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/create_timer_ros.h>
+#include <tf2_ros/message_filter.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <visualization_msgs/msg/marker_array.hpp>
+
+#include <beluga_april_tag_adapter_msgs/msg/feature_detections.hpp>
```

Comment:
This dependency is missing in package.xml/CMakelist.txt and it's failing to build for me due to this.

Reply:
My bad. Read `april_tag` and I was like, this is unused here. It's a bit weird to have a `FeatureDetections` message living in a `beluga_april_tag_adapter_msgs` package, but I'm not making this PR any bigger. We can refactor that later.

Reply:
Restored the build in https://github.com/Ekumen-OS/beluga-demos/pull/8/commits/83fae496e0f3a96bdb42cfa7625a0e217d47897f.

Reply:
> My bad. Read `april_tag` and I was like, this is unused here. FWIW it's a bit weird to have a `FeatureDetections` message living in a `beluga_april_tag_adapter_msgs` package, but I'm not making this PR any bigger. We can refactor that later.

Indeed, it's repurposing the message, so the name is misleading. That needs a cleanup.